pkg/gardenlet/controller/seed/seed_control.go

```diff
@@ -184,13 +184,6 @@ func (c *defaultControl) ReconcileSeed(obj *gardencorev1beta1.Seed, key string)
 			return nil
 		}
 
-		if seed.Spec.Backup != nil {
-			if err := deleteBackupBucketInGarden(ctx, gardenClient.Client(), seed); err != nil {
-				seedLogger.Error(err.Error())
-				return err
-			}
-		}
```

Comment:
I don't understand why the `BackupBucket` is no longer deleted here when the seed's deletion timestamp is set. Why was this code removed / how is the resource deleted now?

Reply:
BB deletion  has to be done manually. 

Before Seed deletion was blocked by the admission plugin if a BB existed. So this code did not make sense in my opinion as there can never be a BB . 

With this PR the Seed deletion is not blocked any more when there is a BB (Seed has a deletion timestamp).
This is why this code has to be removed, otherwise when deleting a Seed with BB, the BB will be deleted automatically.

Reply:
There should be nothing that needs to be done "manually" in Gardener, at least not on purpose. I agree that it should be allowed to set the deletion timestamp on the `Seed` even if there is still a `BackupBucket`, but I still don't get why we have to remove the automatic `BackupBucket` deletion in the seed controller once the object has a deletion timestamp.

Reply:
I think the reasoning was to protect existing Shoot backups from accidental and indirect deletion via the Seed. Right @vpnachev ? 
Is there any value in retaining Backup entries of already deleted Shoots?

Reply:
I can't follow. My question was:

> but I still don't get why we have to remove the automatic BackupBucket deletion in the seed controller once the `Seed` object has a deletion timestamp.

Again: Why did you remove this code?

> Is there any value in retaining Backup entries of already deleted Shoots?

Not sure how this is related now, but yes, sometimes operators want to keep the shoot's etcd backups even after the shoot has been deleted (for some grace period, maybe 2 weeks or so).

Reply:
> Again: Why did you remove this code?

Otherwise this PR introduces a  change of behaviour towards BB deletion.
If this code is removed, a BB is deleted automatically when a Seed has a deletion timestamp.
Previously you had to manually delete the BB before the Seed deletion was allowed by the AdmissionPlugin.

Reply:
Now that I established why I deleted the code, the other question is, if we want that manual BB deletion step in the first place. 
And because @vpnachev introduced the admission plugin validation, I wanted to ask him what his original reason was.

Reply:
> If this code is removed, a BB is deleted automatically when a Seed has a deletion timestamp.
Previously you had to manually delete the BB before the Seed deletion was allowed by the AdmissionPlugin.

Well, this is how it was before @vpnachev introduced the validation. IMO, this is completely fine. I suggest to add this code again and only remove the validation.

> if we want that manual BB deletion step in the first place.

See above: If we can avoid then we don't want any manual step at all anywhere.

Reply:
> See above: If we can avoid then we don't want any manual step at all anywhere.

I think that depends on the following:
> Is there any value in retaining Backup entries (&Bucket) of already deleted Shoots?

Yes automation is great, but  before I remove it, I want to understand why @vpnachev introduced the safeguard that the BB has to be manually deleted. 
We could also automate all Shoot cluster deletion when the Seed resource is deleted (without checking for existing Shoot clusters) - but we choose to not do it for safety reasons right?

Reply:
Excuse me for the delay.

To be honest, after some time I think we should remove the validation for `BackupBuckets` and let the seed controller delete it/them.

Reply:
Alright, thanks. 

> remove the validation for BackupBuckets and let the seed controller delete them.

done. PTAL